frontend/src/scene/main-view/MainView.js

```diff
@@ -32,6 +34,17 @@ class MainView extends React.Component {
                     show={this.state.showCourseBalance}
                     onClear={() => this.setState({ showCourseBalance: false })}
                 />
+                <PaymentView
+                    show={
+                        this.props.location.pathname === '/payment-complete' ||
+                        this.state.showPaymentConfirmationModal
+                    }
+                    params={this.props.location.search}
+                    hidePaymentModal={() => {
+                        this.props.location.pathname = '';
```

Comment:
the recommended way to do this is to render a <Redirect /> component in the modal.

Reply:
fixed as you suggested.